webui/react/src/hooks/usePermissions.ts

```diff
@@ -75,19 +75,42 @@ interface PermissionsHook {
   canViewGroups: boolean;
   canViewWorkspace: (arg0: WorkspacePermissionsArgs) => boolean;
   canViewWorkspaces: boolean;
+  loading: boolean;
 }
 
 const usePermissions = (): PermissionsHook => {
+  const rbacEnabled = useFeature().isOn('rbac');
+  const rbacAllPermission = useFeature().isOn('mock_permissions_all');
+  const rbacReadPermission = useFeature().isOn('mock_permissions_read') || rbacAllPermission;
+
   const loadableCurrentUser = useCurrentUser();
   const user = Loadable.match(loadableCurrentUser, {
     Loaded: (cUser) => cUser,
     NotLoaded: () => undefined,
   });
-  const rbacEnabled = useFeature().isOn('rbac');
-  const rbacAllPermission = useFeature().isOn('mock_permissions_all');
-  const rbacReadPermission = useFeature().isOn('mock_permissions_read') || rbacAllPermission;
-  const userAssignments = Loadable.getOrElse([], useUserAssignments());
-  const userRoles = Loadable.getOrElse([], useUserRoles());
+  const loadableUserAssignments = useUserAssignments();
+  const userAssignments = Loadable.match(loadableUserAssignments, {
+    Loaded: (uAssignments) => uAssignments,
+    NotLoaded: () => [],
+  });
+  const loadableUserRoles = useUserRoles();
+  const userRoles = Loadable.match(loadableUserRoles, {
+    Loaded: (uRoles) => uRoles,
+    NotLoaded: () => [],
+  });
+
+  const [loading, setLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    if (
+      !Loadable.isLoading(loadableCurrentUser) &&
```

Comment:
Does this need to be state?

Reply:
i think it does -- tried `useRef` but had issues with it not triggering re-render in `<Page>` component, so spinner doesn't get replaced with content as desired.

Reply:
Why can't you just return `Loadable.isLoading(loadableCurrentUser) || ...`?

Reply:
realized i could do that right after i commented :) updating...